pdu/ShortMessage.go

```diff
@@ -66,6 +66,17 @@ func (c *ShortMessage) SetMessageWithEncoding(message string, enc data.Encoding)
 			c.message = message
 			c.enc = enc
 		}
+
```

Comment:
Wouldn't this added code block be better place inside the Encode func for GSM in codings.go ?

I'm just asking.....

Reply:
Thanks for the review @laduchesneau. Actually I tried to move the logic there into the (c *gsm7bit) Encode(str string) method. But the problem is that, this tail cleaning operation is dependent on multi-part / single part type of a message. If Encode() is triggered from a single message then calculation is based on **(nSeptet+1) % 8 == 0**, because we do not shift the bits left when UDH is not present in a message stream, meaning for a single part message. If Encode() is triggered from a multi-part message (pls see EncodeSplit->shiftBitsLeftOne), then the calculation needs to be based on **nSeptet%8 == 0** . Since Encode() is a method that implements the EncDec interface like others, we can not parameterize it. For this reason, I though it can be placed in the SetMessageWithEncoding method. I am not super happy with the placement, but couldn't find any other alternative :) What do you think? Any other place suggestion to move or verbose Encode() interface implementation or leave it as it is?